### PR TITLE
Cover UnsubscribeWayPoints and UpdateTurnList by Unit Tests.

### PR DESCRIPTION
**Pull request by LevchenkoS**
Has been covered:
- `UnSubscribeWayPointsRequest`;
- `UnsubscribeWayPointsResponse`;
- `UpdateTurnListRequest`;
- `UpdateTurnListResponse`.

Related to: [APPLINK-25921](https://adc.luxoft.com/jira/browse/APPLINK-25921)

@AByzhynar, @okozlovlux, @Kozoriz, @LuxoftAKutsan, @vlantonov, @wolfylambova22, @VVeremjova , @pvvasilev, please review.

### CODE REVIEW

File: src/components/application_manager/test/commands/mobile/unsubscribe_way_points_request_test.cc [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+      .WillOnce(Return(true));
+
+  EXPECT_CALL(app_mngr_,
+              ManageHMICommand(HMIResultCodeIs(
+                  hmi_apis::FunctionID::Navigation_UnsubscribeWayPoints)));
+
+  command_->Run();
+}
+
+TEST_F(UnSubscribeWayPointsRequestTest, OnEvent_UNSUCCESS) {
```
Comment (okozlovlux) [inline src/components/application_manager/test/commands/mobile/unsubscribe_way_points_request_test.cc]: @LevchenkoS maybe better to name it like "OnEvent_UnknownEvent_UNSUCCESS" ?

File: src/components/application_manager/test/commands/mobile/update_turn_list_request_test.cc [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+
+  MockAppPtr mock_app(CreateMockApp());
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app));
+
+  EXPECT_CALL(app_mngr_, GetPolicyHandler())
+      .WillOnce(ReturnRef(*static_cast<policy::PolicyHandlerInterface*>(NULL)));
+
+  const mobile_result::eType kExpecterResult = mobile_result::INVALID_ENUM;
+  EXPECT_CALL(*mock_message_helper_, ProcessSoftButtons(_, _, _, _))
```
Comment (okozlovlux) [inline src/components/application_manager/test/commands/mobile/update_turn_list_request_test.cc]: @LevchenkoS we don't care about all four args?

File: src/components/application_manager/test/commands/mobile/unsubscribe_way_points_request_test.cc (hunk 1 above)
Comment (AByzhynar) [inline src/components/application_manager/test/commands/mobile/unsubscribe_way_points_request_test.cc]: +1 to @okozlovlux

**Comment by AByzhynar** [comment]
@LevchenkoS Please find out why Travis build was not run on your PR. It's also mandatory to check it on Linux .

File: src/components/application_manager/test/commands/mobile/update_turn_list_request_test.cc [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+  command_->Run();
+}
+
+TEST_F(UpdateTurnListRequestTest, Run_NoTurnList_UNSUCCESS) {
+  MockAppPtr mock_app(CreateMockApp());
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app));
+
+  EXPECT_CALL(app_mngr_, GetPolicyHandler())
+      .WillOnce(ReturnRef(*static_cast<policy::PolicyHandlerInterface*>(NULL)));
```
Comment (vlantonov) [inline src/components/application_manager/test/commands/mobile/update_turn_list_request_test.cc]: @LevchenkoS Do you really dereference a NULL pointer? If so this means undefined behavior!

File: src/components/application_manager/test/commands/mobile/update_turn_list_request_test.cc [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+                 [am::strings::turn_icon][am::strings::value] =
+                     "valid_turn_icon";
+  (*command_msg_)[am::strings::msg_params][am::strings::soft_buttons] = 0;
+
+  MockAppPtr mock_app(CreateMockApp());
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app));
+
+  EXPECT_CALL(app_mngr_, GetPolicyHandler())
+      .WillOnce(ReturnRef(*static_cast<policy::PolicyHandlerInterface*>(NULL)));
```
Comment (vlantonov) [inline src/components/application_manager/test/commands/mobile/update_turn_list_request_test.cc]: @LevchenkoS Do you really dereference a NULL pointer? If so this means undefined behavior!

File: src/components/application_manager/test/commands/mobile/update_turn_list_request_test.cc [hunk 5]
```diff
@@ -0,0 +1,289 @@
...
+  (*command_msg_)[am::strings::msg_params][am::strings::turn_list][0]
+                 [am::strings::turn_icon][am::strings::value] =
+                     "valid_turn_icon";
+
+  MockAppPtr mock_app(CreateMockApp());
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app));
+
+  EXPECT_CALL(app_mngr_, GetPolicyHandler())
+      .WillOnce(ReturnRef(*static_cast<policy::PolicyHandlerInterface*>(NULL)));
```
Comment (vlantonov) [inline src/components/application_manager/test/commands/mobile/update_turn_list_request_test.cc]: @LevchenkoS Do you really dereference a NULL pointer? If so this means undefined behavior! Under Linux - a crash!

File: src/components/application_manager/test/commands/mobile/update_turn_list_request_test.cc [hunk 6]
```diff
@@ -0,0 +1,289 @@
...
+
+  const mobile_result::eType kExpecterResult = mobile_result::INVALID_ENUM;
```
Comment (wolfylambova22) [inline src/components/application_manager/test/commands/mobile/update_turn_list_request_test.cc]: @LevchenkoS is this suppesed to be kExpectedResult or am I confused by the name

Comment (okozlovlux) [inline src/components/application_manager/test/commands/mobile/update_turn_list_request_test.cc]: @LevchenkoS kExpecte**d**Result
Typo

Comment (LevchenkoS) [inline src/components/application_manager/test/commands/mobile/update_turn_list_request_test.cc]: @okozlovlux, @wolfylambova22, fixed in 5b40a53.

File: src/components/application_manager/test/commands/mobile/update_turn_list_request_test.cc (hunk 5 above)
Comment (LevchenkoS) [inline src/components/application_manager/test/commands/mobile/update_turn_list_request_test.cc]: @vlantonov, fixed in 5b40a53.

File: src/components/application_manager/test/commands/mobile/update_turn_list_request_test.cc (hunk 4 above)
Comment (LevchenkoS) [inline src/components/application_manager/test/commands/mobile/update_turn_list_request_test.cc]: @vlantonov, fixed in 5b40a53.

File: src/components/application_manager/test/commands/mobile/update_turn_list_request_test.cc (hunk 3 above)
Comment (LevchenkoS) [inline src/components/application_manager/test/commands/mobile/update_turn_list_request_test.cc]: @vlantonov, fixed in 5b40a53.

File: src/components/application_manager/test/commands/mobile/update_turn_list_request_test.cc (hunk 2 above)
Comment (LevchenkoS) [inline src/components/application_manager/test/commands/mobile/update_turn_list_request_test.cc]: @vlantonov, fixed in 5b40a53.

File: src/components/application_manager/test/commands/mobile/unsubscribe_way_points_request_test.cc (hunk 1 above)
Comment (LevchenkoS) [inline src/components/application_manager/test/commands/mobile/unsubscribe_way_points_request_test.cc]: @okozlovlux, @AByzhynar, fixed in 5b40a53.

**Comment by okozlovlux** [comment]
@LevchenkoS , reviewed

**Comment by Kozoriz** [comment]
Reviewed

**Comment by vlantonov** [comment]
@LevchenkoS Reviewed

**Comment by wolfylambova22** [comment]
Reviewed.

**Comment by pvvasilev** [comment]
Reviewed.

File: src/components/application_manager/test/commands/mobile/update_turn_list_request_test.cc [hunk 7]
```diff
@@ -0,0 +1,289 @@
...
+      : mock_message_helper_(am::MockMessageHelper::message_helper_mock())
+      , command_msg_(CreateMessage(smart_objects::SmartType_Map))
+      , command_(CreateCommand<UpdateTurnListRequest>(command_msg_)) {
+    (*command_msg_)[am::strings::params][am::strings::connection_key] =
+        kConnectionKey;
+    (*command_msg_)[am::strings::params][am::strings::function_id] =
+        kFunctionId;
+  }
+
+  am::MockMessageHelper* mock_message_helper_;
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/commands/mobile/update_turn_list_request_test.cc]: @LevchenkoS  can we hold it not as raw pointer? Who is responsible for life cycle of this pointer?

File: src/components/application_manager/test/commands/mobile/update_turn_list_request_test.cc [hunk 8]
```diff
@@ -0,0 +1,289 @@
...
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app));
+
+  EXPECT_CALL(app_mngr_, GetPolicyHandler())
+      .WillOnce(ReturnRef(*static_cast<policy::PolicyHandlerInterface*>(NULL)));
+
+  EXPECT_CALL(*mock_message_helper_, ProcessSoftButtons(_, _, _, _))
+      .WillOnce(Return(mobile_result::SUCCESS));
+
+  EXPECT_CALL(*mock_message_helper_, VerifyImage(_, _, _))
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/commands/mobile/update_turn_list_request_test.cc]: @LevchenkoS do you care about image to verify?

File: src/components/application_manager/test/commands/mobile/update_turn_list_request_test.cc [hunk 9]
```diff
@@ -0,0 +1,289 @@
...
+  (*command_msg_)[am::strings::msg_params][am::strings::soft_buttons] = 0;
+
+  MockAppPtr mock_app(CreateMockApp());
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app));
+
+  EXPECT_CALL(app_mngr_, GetPolicyHandler())
+      .WillOnce(ReturnRef(*static_cast<policy::PolicyHandlerInterface*>(NULL)));
+
+  EXPECT_CALL(*mock_message_helper_, ProcessSoftButtons(_, _, _, _))
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/commands/mobile/update_turn_list_request_test.cc]: @LevchenkoS do you care about buttons to process?

File: src/components/application_manager/test/commands/mobile/update_turn_list_request_test.cc (hunk 8 above)
Comment (LevchenkoS) [inline src/components/application_manager/test/commands/mobile/update_turn_list_request_test.cc]: @LuxoftAKutsan, it has been fixed in 621e7b7.

File: src/components/application_manager/test/commands/mobile/update_turn_list_request_test.cc (hunk 9 above)
Comment (LevchenkoS) [inline src/components/application_manager/test/commands/mobile/update_turn_list_request_test.cc]: @LuxoftAKutsan, it has been fixed in 621e7b7.

File: src/components/application_manager/test/commands/mobile/update_turn_list_request_test.cc (hunk 7 above)
Comment (LevchenkoS) [inline src/components/application_manager/test/commands/mobile/update_turn_list_request_test.cc]: In constructor of `UpdateTurnListRequestTest` `mock_message_helper_` get pointer to the static variable hidden inside of `MockMessageHelper` class.

Comment (LuxoftAKutsan) [inline src/components/application_manager/test/commands/mobile/update_turn_list_request_test.cc]: @LevchenkoS why you don't handle it by reference?

File: src/components/application_manager/test/commands/mobile/unsubscribe_way_points_request_test.cc (hunk 1 above)
Comment (AByzhynar) [inline src/components/application_manager/test/commands/mobile/unsubscribe_way_points_request_test.cc]: @LevchenkoS Please rename all test cases according their logic. Include incoming parameters into test case name.

File: src/components/application_manager/test/commands/mobile/update_turn_list_request_test.cc (hunk 7 above)
Comment (LevchenkoS) [inline src/components/application_manager/test/commands/mobile/update_turn_list_request_test.cc]: @LuxoftAKutsan, fixed in 7fd9bab.

File: src/components/application_manager/test/commands/mobile/update_turn_list_request_test.cc [hunk 10]
```diff
@@ -0,0 +1,289 @@
...
+using ::testing::Return;
+using ::testing::ReturnRef;
+
+using am::commands::UpdateTurnListRequest;
+using am::commands::MessageSharedPtr;
+using application_manager_test::MockHMICapabilities;
+
+namespace {
+const uint32_t kConnectionKey = 3u;
+const uint32_t kFunctionId = 2;
```
Comment (LevchenkoS) [inline src/components/application_manager/test/commands/mobile/update_turn_list_request_test.cc]: @LuxoftAKutsan, fixed in 7fd9bab.

**Comment by Kozoriz** [comment]
Reviewed

File: src/components/application_manager/test/commands/mobile/unsubscribe_way_points_request_test.cc (hunk 1 above)
Comment (LevchenkoS) [inline src/components/application_manager/test/commands/mobile/unsubscribe_way_points_request_test.cc]: @okozlovlux, @AByzhynar, fixed in ff89c63.

**Comment by AByzhynar** [comment]
Reviewed

**Comment by VVeremjova** [comment]
Reviewed

**Pull request merged**